packages/token-bridge-sdk/src/util/graph.ts

```diff
@@ -51,6 +67,40 @@ const networkIDAndLayerToClient = (networkID: string, layer: 1 | 2) => {
   }
 }
 
+export const getNodes = async (
+  networkID: string,
+  minAfterSendCount = 0,
+  offset = 0
+): Promise<NodeDataResult[]> => {
+  const client = networkIDAndLayerToClient(networkID, 1)
+  const res = await client.query({
+    query: gql`
+    {
+      nodes( 
+        orderBy: afterSendCount
+        orderDirection: asc
+        where:{ afterSendCount_gte: ${minAfterSendCount}} 
+        first: 1000,
+        skip: ${offset}
+
+      ){
+        afterSendCount,
+        timestampCreated,
+        blockCreatedAt,
+        id
+      }
+    }
+    `
+  })
+  const nodes = res.data.nodes as NodeDataResult[]
+  if (nodes.length === 0) {
+    return nodes
+  } else {
+    return nodes.concat(
+      await getNodes(networkID, minAfterSendCount, offset + nodes.length)
```

Comment:
could turn this into a do/while loop to avoid the recursive call overhead, but this approach is also neat

Reply:
yeah idk I just think it's neat

Reply:
I agree, wish we had tail recursion optimisation :(